src/nccl_ofi_net.c

```diff
@@ -965,26 +1009,30 @@ static ncclResult_t ofi_process_cq(nccl_ofi_t *nccl_ofi_comp)
 			} else if (OFI_UNLIKELY(rc < 0)) {
 				NCCL_OFI_WARN("Unable to read from fi_cq_readerr. RC: %zd. Error: %s",
 					     rc,
-					     fi_cq_strerror(cq,
-						err_buffer.prov_errno,
-						err_buffer.err_data, NULL, 0));
+					     fi_strerror(-rc));
 				ret = ncclSystemError;
 				goto exit;
 			}
 
-			/* TODO: Add debug log to dump failed request details */
 			req = container_of(err_buffer.op_context,
 					   nccl_ofi_req_t, ctx);
 			req->state = NCCL_OFI_REQ_ERROR;
 			req->size = err_buffer.len;
```

Comment:
Shouldn't this come after the print statement as otherwise we will always have request state has `NCCL_OFI_REQ_ERROR`?

Reply:
Would we want to report previous state?  If we do, won't it always be `NCCL_OFI_REQ_PENDING`?  We are polling a cq, so what else would it be?

...except that we seem to have a bug right now: we don't use the `PENDING` status and leave the request as `CREATED` after we start it.

Reply:
I actually thought about this question when writing the code and decided to write it this way.  Reasoning: we may want to add code to dump the pending request queue at some point and it would be helpful to print actual rather than previous status here.  Also, the completed length that is reported here is useful to report, and it seemed weird to update one field by not another.

Reply:
Yeah we have that bug to change the status correctly. I remember fixing it in one of my local branches. We should fix that separately.

This dump is for erroneous requests and my suggestion was to get the previous state of the request before we alter it and mark as finished in error. Maybe we should print the completed length separately (like `err_buffer.err`)  and get the original request information from the dump? I agree that most of the requests will be in pending but such dumps may help us to understand our code bugs ;)